cg/services/run_devices/pacbio/metrics_parser/utils.py

```diff
@@ -32,12 +35,39 @@ def _parse_report_to_model(report_file: Path, data_model: Type[BaseMetrics]) ->
     parsed_json: dict = ReadFile.read_file[FileFormat.JSON](report_file)
     if data_model == SmrtlinkDatasetsMetrics:
         return data_model.model_validate(parsed_json[0], from_attributes=True)
-    metrics: list[dict[str, Any]] = parsed_json.get("attributes")
-    data: dict = {report_field["id"]: report_field["value"] for report_field in metrics}
+    metrics: list[dict[str, Any]] = parsed_json.get(MetricsFileFields.ATTRIBUTES)
+    data: dict = {
+        report_field["id"]: report_field[MetricsFileFields.VALUE_ONE] for report_field in metrics
+    }
     return data_model.model_validate(data, from_attributes=True)
 
 
 def get_parsed_metrics_from_file_name(metrics_files: list[Path], file_name: str) -> BaseMetrics:
     report_file: Path = get_file_with_pattern_from_list(files=metrics_files, pattern=file_name)
     data_model: Type[BaseMetrics] = _get_data_model_from_pattern(pattern=file_name)
     return _parse_report_to_model(report_file=report_file, data_model=data_model)
+
+
+def _parse_sample_data(sample_data: list[dict[str, Any]]) -> list[SampleMetrics]:
```

Comment:
I find this method and the whole parsing quite hard to read, and quite dependent on me having knowledge of how the file we are trying to parse is structured. With my brain being a bit mushy, would it be possible to use the csv module and some syntax similar to this?

<img width="748" alt="image" src="https://github.com/user-attachments/assets/c5b3930f-9616-4549-9e7f-9a035475d901">

Reply:
I will leave it as it is and we can have a discussion in the development branch when more changes are in place